frontend/fleet/helpers.ts

```diff
@@ -373,6 +433,22 @@ export const secondsToHms = (d: number): string => {
   return hDisplay + mDisplay + sDisplay;
 };
 
+export const secondsToDhms = (d: number): string => {
+  if (d === 604800) {
+    return "1 week";
+  }
+  const day = Math.floor(d / (3600 * 24));
+  const h = Math.floor((d % (3600 * 24)) / 3600);
+  const m = Math.floor((d % 3600) / 60);
+  const s = Math.floor(d % 60);
+
+  const dDisplay = day > 0 ? day + (day === 1 ? " day " : " days ") : "";
+  const hDisplay = h > 0 ? h + (h === 1 ? " hour " : " hours ") : "";
+  const mDisplay = m > 0 ? m + (m === 1 ? " minute " : " minutes ") : "";
+  const sDisplay = s > 0 ? s + (s === 1 ? " second" : " seconds") : "";
+  return dDisplay + hDisplay + mDisplay + sDisplay;
+};
+
```

Comment:
This would be a great candidate for  a unit test.

Reply:
Done! ✅